### PR TITLE
Issue 338: CRD template refactor to accommodate different api versions

### LINKED ISSUE #338: Fix unknown field problems after change to CRD apiVersion check
### Description

The CRD apiVersion check introduced a validation problem due to changes on the resource type schema

### Importance

zookeeper-operator chart can't be installed on EKS versions supporting apiextensions.k8s.io/v1

### Location

https://github.com/lspecian-olx/zookeeper-operator/blob/master/charts/zookeeper-operator/templates/zookeeper.pravega.io_zookeeperclusters_crd.yaml#L3

### Suggestions for an improvement

The template needs to accommodate both api versions apiextensions.k8s.io/v1 and apiextensions.k8s.io/v1beta1 in order to continue to provide support for eks versions without support for apiextensions.k8s.io/v1

### PR DESCRIPTION
**Pull request by lspecian-olx**
### Change log description
In order to accommodate the both api versions schemas apiextensions.k8s.io/v1 and apiextensions.k8s.io/v1beta1 
- moved openAPIV3Schema as a named template in _helpers.tpl since the schema change moved the object 
- the schema change also moved the object additionalPrinterColumns which needed two versions as named templates since the attribute JSONPath was changed to jsonPath

### Purpose of the change

Closes #338 

### What the code does

The code is checking .Capabilities.APIVersions.Has "apiextensions.k8s.io/v1" and adapting the template output accordingly to the support for the apiVersion 

### How to verify it

Chart is able to install in eks clusters that support both apiVersions apiextensions.k8s.io/v1 and apiextensions.k8s.io/v1beta1

### CODE REVIEW

**Comment by anishakj** [comment]
@lspecian-olx , how you have generated  crd file? Is there any tools used

**Comment by lspecian-olx** [comment]
> @lspecian-olx , how you have generated crd file? Is there any tools used

hello, I used only helm templates, not sure if I understood the question

**Comment by anishakj** [comment]
> > @lspecian-olx , how you have generated crd file? Is there any tools used
> 
> hello @anishakj, I used only helm templates, not sure if I understood the question

Have you used any commands like `operatpr-sdk generate crds` ,but have seen that operator-sdk generate crd with `v1beta1` version

**Comment by anishakj** [comment]
@lspecian-olx, could you please resolve the conflict?

**Comment by anishakj** [comment]
@lspecian-olx  Looks like there is parse error after conflicts are resolved. Getting below error while try to install

```
Error: YAML parse error on zookeeper-operator/templates/zookeeper.pravega.io_zookeeperclusters_crd.yaml: error converting YAML to JSON: yaml: line 70: could not find expected ':'
```

**Comment by hansh0801** [comment]
thanks!

**Review by SrishT — CHANGES_REQUESTED** [review]
Just a very minor comment needs to be addressed, otherwise LGTM

File: charts/zookeeper-operator/templates/_helpers.tpl [hunk 1]
```diff
@@ -75,8 +74,7 @@ additionalPrinterColumns:
       name: Replicas
       type: integer
     - JSONPath: .status.readyReplicas
-      description: The number of ZooKeeper servers in the ensemble that are in a Ready
-        state
+      description: The number of ZooKeeper servers in the ensemble that are in a Ready sstate
```
Comment (anishakj) [inline charts/zookeeper-operator/templates/_helpers.tpl:77]: Could you please correct the typo `sstate`

**Review by anishakj — CHANGES_REQUESTED** [review]

**Review by anishakj — APPROVED** [review]
LGTM

**Review by SrishT — APPROVED** [review]
LGTM

**Pull request merged**